### ISSUE TITLE
Information on using a newer languagetool build

### ISSUE THREAD

**Issue by nighthawk9** [sylvainhalle/textidote#193]
Hello sylvainhalle,

I just started using your tool for editing LaTeX documents and something that I thought I didn't need until now! Thank you for your time and effort in developing the tool.

In short, I am unfamiliar with Java and I was looking to use a newer build of the languagetools utility in textidote. I spent a fair amount of time trying to figure out exactly what is required to build with a new version, but have been unsuccessful.

I would greatly appreciate any direction or documentation on updating to a new version of languagetools, and could help in offering a new LT bundle?

Any information is appreciated!

Thank you,

nighthawk9

**Comment by sylvainhalle** [comment]
Hi,

In the current way TeXtidote is built, upgrading to a new version of LanguageTool (LT) requires one to build a stand-alone or "fat" JAR with all its dependencies bundled in a single file. A long time ago, I succeeded in creating such a fat JAR with the sources of LT 4.6, and it is included as a dependency within TeXtidote.

https://github.com/sylvainhalle/textidote/blob/769d6984aa070596f69cc2cfba7be1619a4d0e1e/config.xml#L77

Upgrading to a newer LT should "just" be a matter of replacing this file with one that would be built from a more recent version. However, the process was mostly manual trial-and-error fiddling (especially since one of the files within the fat JAR also [had to be manually edited after compiling](https://github.com/sylvainhalle/languagetool/issues/1)), so I postponed doing it again with another version of LT ever since.

Upgrading LT has been promised in a number of issues (notably #188); last year @bratekarate opened issue #136 with instructions on how to build a new fat JAR, however, at first I did not succeed in reproducing them and then had to leave the open due to lack of time.

So as you see, it all hinges on the existence of a newer fat JAR for LanguageTool that could replace the old one, which is more complicated than it looks.
